### ISSUE TITLE
bug(android/engine) Android system keyboard popup key misaligned

### ISSUE THREAD

**Issue by darcywong00** [keymanapp/keyman#4019]
During testing in #3989 I noticed the popup keys  appear at inconsistent heights

### Configuration
Keyman for Android 14.0.191 alpha
Motorola G5S+, Android 8.1 Oreo

sil_eruo_latin with suggestion banner disabled

### Screenshots
longpress on <kbd>u</kbd>

In-app popup keys appear at their expected height (above the OSK)
![inapp](https://user-images.githubusercontent.com/7358010/100574634-05cf5f80-330d-11eb-98e8-6c73a4056061.jpg)

System keyboard popup keys appear below expected behavior (popup aligning to <kbd>n</kbd> key.
I would have expected the popups to match the in-app alignment
![sys](https://user-images.githubusercontent.com/7358010/100574718-33b4a400-330d-11eb-87b0-ee3cfb904c2d.jpg)

**Comment by jahorton** [comment]
Why is this labeled with `web/`?  Web is not responsible for how Android displays the popup keys.

**Comment by jahorton** [comment]
There's a _chance_ that the position of the key is incorrectly reported when in the system keyboard context, but even then... it's likely still more of an Android issue.

**Comment by darcywong00** [comment]
My bad. I tracked this down to system keyboard parameters in #3009

14.0.52 alpha - system keyboard popups are good
14.0.53 alpha - system keyboard popups are misaligned.

**Comment by jahorton** [comment]
So that was the WeChat fix, right?  Which means we can't simply revert that PR.

**Comment by darcywong00** [comment]
> So that was the WeChat fix, right?

Also seen in Telegram app. imho, the emojis filling the screen impacts a lot less users than mis-aligned system keyboard  popups where the popup can be right under the key you're holding.

**Comment by jahorton** [comment]
It doesn't look like https://github.com/keymanapp/keyman/issues/1870 (the old WeChat issue) was in regard to emojis.  My read of it was that _any_ use of the Keyman keyboard within that app was affected.

**Comment by darcywong00** [comment]
Jotting down notes.

In the Telegram app, clicking the emoji button dismisses the keyboard:
```
InputMethodManager imm = (InputMethodManager) view.getContext().getSystemService(Context.INPUT_METHOD_SERVICE);
...
imm.hideSoftInputFromWindow(view.getWindowToken(), 0);
```

but this doesn't trigger Keyman keyboard event handlers (onKeyboardDismissed)

Reference

https://github.com/DrKLO/Telegram/blob/master/TMessagesProj/src/main/java/org/telegram/messenger/AndroidUtilities.java

**Comment by darcywong00** [comment]
Found a [SO](https://stackoverflow.com/questions/52929466/popupwindow-getting-clipped-on-custom-keyboard-for-android-api-28/53326786) page that implies Android might be limiting the popups to the Keyboard region. (similar to iOS?)

**Comment by jahorton** [comment]
That would definitely match the screenshots seen above; the subkeys seem y-shifted down to fit within the keyboard's area.

**Comment by mcdurdin** [comment]
The answers on that SO page give us some avenues to resolve this (also the suggestion that the issue is resolved in API 29?)

**Comment by mcdurdin** [comment]
> the suggestion that the issue is resolved in API 29

Tested against API 29 and it made no difference. Now need to investigate the other solutions.

**Comment by darcywong00** [comment]
on API 21, the popup keys appear in the correct location, but only the keys within the banner are clickable
![api 21](https://user-images.githubusercontent.com/7358010/102303759-8093ae00-3f8e-11eb-964b-9ffa9d2635ea.png)

**Comment by darcywong00** [comment]
Fixed in 14.0 beta by #4254
